src/views/PeopleSearch/index.js

```diff
@@ -287,83 +281,70 @@ class PeopleSearch extends Component {
     this.setState({
-      firstNameSearchValue: e.target.value,
+      searchValues: { ...this.state.searchValues, firstName: e.target.value },
     });
   };
   handleLastNameInputChange = (e) => {
     this.setState({
-      lastNameSearchValue: e.target.value,
+      searchValues: { ...this.state.searchValues, lastName: e.target.value },
     });
   };
   handleMajorInputChange = (e) => {
     this.setState({
-      majorSearchValue: e.target.value,
+      searchValues: { ...this.state.searchValues, major: e.target.value },
     });
   };
   handleMinorInputChange = (e) => {
     this.setState({
-      minorSearchValue: e.target.value,
+      searchValues: { ...this.state.searchValues, minor: e.target.value },
     });
   };
   handleHallInputChange = (e) => {
     this.setState({
-      hallSearchValue: e.target.value,
+      searchValues: { ...this.state.searchValues, hall: e.target.value },
     });
   };
   handleClassTypeInputChange = (event) => {
     this.setState({
-      classTypeSearchValue: event.target.value,
+      searchValues: { ...this.state.searchValues, classType: event.target.value },
     });
   };
   handleHomeCityInputChange = (e) => {
     this.setState({
-      homeCitySearchValue: e.target.value,
+      searchValues: { ...this.state.searchValues, homeCity: e.target.value },
     });
   };
   handleStateInputChange = (e) => {
     this.setState({
-      stateSearchValue: e.target.value,
+      searchValues: { ...this.state.searchValues, state: e.target.value },
     });
   };
   handleCountryInputChange = (e) => {
     this.setState({
-      countrySearchValue: e.target.value,
+      searchValues: { ...this.state.searchValues, country: e.target.value },
     });
   };
   handleDepartmentInputChange = (e) => {
     this.setState({
-      departmentSearchValue: e.target.value,
+      searchValues: { ...this.state.searchValues, department: e.target.value },
     });
   };
   handleBuildingInputChange = (e) => {
     this.setState({
-      buildingSearchValue: e.target.value,
+      searchValues: { ...this.state.searchValues, building: e.target.value },
     });
   };
 
-  handleChangeIncludeAlumni() {
-    this.setState({ includeAlumni: !this.state.includeAlumni });
-  }
-
   async search() {
-    if (
-      this.state.includeAlumni === false &&
-      this.state.firstNameSearchValue === '' &&
-      this.state.lastNameSearchValue === '' &&
-      this.state.classTypeSearchValue === '' &&
-      this.state.majorSearchValue === '' &&
-      this.state.minorSearchValue === '' &&
-      this.state.hallSearchValue === '' &&
-      this.state.classTypeSearchValue === '' &&
-      this.state.homeCitySearchValue === '' &&
-      this.state.stateSearchValue === '' &&
-      this.state.countrySearchValue === '' &&
-      this.state.departmentSearchValue === '' &&
-      this.state.buildingSearchValue === ''
-    ) {
-      // do not search
+    if (!Object.values(this.state.searchValues).some((x) => x !== '' && x !== false)) {
```

Comment:
While this is a good explicit check, `''` and `false` are both falsy values, so this would work just as well as `some((x) => x)